File: volume/tasks-manager/app/models/task_list.rb
class TaskList < ApplicationRecord
  belongs_to :user, optional: false
  has_many :task_in_lists, dependent: :destroy

  validates :name, presence: true, uniqueness: { scope: :user_id }
  validates :frequence_type, presence: true

  enum frequence_type: { daily: 0, monthly: 1, weekly: 2, yearly: 3 }
end

File: volume/tasks-manager-client/src/app/model/tasklist.ts
import { TaskItem } from './task_item';
export class Tasklist {
  id: number;
  name: string;
  description: string;
  frequenceType: string;
  taskItems: TaskItem[];

  constructor(tasklist) {
    this.id = tasklist.id;
    this.name = tasklist.name;
    this.description = tasklist.description;
    this.frequenceType = tasklist.frequence_type;
    this.taskItems = tasklist.task_in_lists.map(
      (taskItem: any) => new TaskItem(taskItem, this)
    );
  }
}
